File: redo/reredo/darwin/progressbar.m
// 14 august 2015
#import "uipriv_darwin.h"

// TODOs:
// - 10.8: increasing value animates just like with Aero

struct uiProgressBar {
	uiDarwinControl c;
	NSProgressIndicator *pi;
};

uiDarwinDefineControl(
	uiProgressBar,							// type name
	uiProgressBarType,						// type function
	pi									// handle
)

void uiProgressBarSetValue(uiProgressBar *p, int value)
{
	if (value < 0 || value > 100)
		complain("value %d out of range in progressbarSetValue()", value);
	[p->pi setDoubleValue:((double) value)];
}

uiProgressBar *uiNewProgressBar(void)
{
	uiProgressBar *p;

	p = (uiProgressBar *) uiNewControl(uiProgressBarType());

	p->pi = [[NSProgressIndicator alloc] initWithFrame:NSZeroRect];
	[p->pi setControlSize:NSRegularControlSize];
	[p->pi setBezeled:YES];
	[p->pi setStyle:NSProgressIndicatorBarStyle];
	[p->pi setIndeterminate:NO];

	uiDarwinFinishNewControl(p, uiProgressBar);

	return p;
}

File: winforms/progressbar.cpp
// 26 november 2015
#include "uipriv_winforms.hpp"

struct uiProgressBar {
	uiWindowsControl c;
	DUMMY dummy;
};

uiWindowsDefineControl(
	uiProgressBar,							// type name
	uiProgressBarType,						// type function
	dummy								// handle
)

void uiProgressBarSetValue(uiProgressBar *p, int n)
{
	// TODO
}

uiProgressBar *uiNewProgressBar(void)
{
	uiProgressBar *p;

	p = (uiProgressBar *) uiNewControl(uiProgressBarType());

	p->dummy = mkdummy(L"uiProgressBar");

	uiWindowsFinishNewControl(p, uiProgressBar, dummy);

	return p;
}
